js/plates-model/model-worker.js

```diff
@@ -56,8 +56,16 @@ function workerFunction () {
 onmessage = function modelWorkerMsgHandler (event) {
   const data = event.data
   if (data.type === 'load') {
-    // Export model to global m variable for convenience.
-    self.m = model = new Model(data.imgData, presets[data.presetName].init)
+    if (data.loadType === 'preset') {
+      // Export model to global m variable for convenience.
+      self.m = model = new Model(data.imgData, presets[data.presetName].init)
+    } else if (data.loadType === 'cloud') {
+      let deserializedModel = Model.deserialize(JSON.parse(data.serializedModel))
+      // The model may have been stored mid-run, so reset it to ensure it is properly initialized
+      deserializedModel.time = 0
+      deserializedModel.stepIdx = 0
+      self.m = model = deserializedModel
```

Comment:
Two details here:
- I'd divide that into two separate handlers, as they don't seem to share anything. E.g. 'loadPreset` and `loadModel` (this handler doesn't seem to have anything related to the cloud and we might use it for another loading, e.g from file or any other source).
- Not sure about reseting time and stepIdx. It's part of the serialized model and my general feeling is that we should preserve that. But on the other hand, we don't display that anywhere. Was it causing any problems?

Reply:
Good point about the handlers! It was causing problems in 2 places, basically things were not getting initialized properly:

https://github.com/concord-consortium/plate-tectonics-3d/blob/master/js/plates-model/model-output.js#L24-L29

https://github.com/concord-consortium/plate-tectonics-3d/blob/master/js/plates-model/model-worker.js#L26-L28

Reply:
Hm, I thought you removed the first check (step < 2) in one of your previous PRs. We should probably do it, as it's an optimization which might cause more problems. The initial snapshot would require a bit more work, though.

OK, since we don't display time anywhere, let's keep it that way for now.